javagen/src/main/resources/MultipartFormDataHelper.java

```diff
@@ -215,7 +216,9 @@ private void appendBytes(byte[] bytes) {
         requestDataStream = new SequenceInputStream(requestDataStream, new ByteArrayInputStream(bytes));
     }
 
+    private static final Pattern REDACT_FILENAME = Pattern.compile("[^\\x20-\\x7E]|\"");
```

Comment:
Can you add a comment why this is needed? Why is the filename redacted?

Reply:
The header allows ASCII, while the filename may include much more.
Client is allowed to change the filename (somehow).

[rfc8187](https://www.ietf.org/rfc/rfc8187.html#section-3.2.3) & [rfc6266](https://www.ietf.org/rfc/rfc6266.html#section-4.3) recommend "filename" be ASCII, and a "filename*" with encoded UTF8.

At present, we hadn't get to the "filename*" yet.